File: programmers/level-3/12900. 2 x n 타일링/solution.cc
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    if (n < 3) return n;
    vector<int> vec = {1, 2};
    for (int i = 2; i < n; i += 1) {
        int t = vec[0];
        vec[0] = vec[1];
        vec[1] = (vec[0] + t) % 1000000007;
    }
    return vec[1];
}

File: programmers/level-3/12900. 2 x n 타일링/solution.js
/**
 * https://programmers.co.kr/learn/courses/30/lessons/12900
 * @param {number} n 
 */
function solution(n) {
    if (n < 3) return n;
    const dp = [1, 2];
    for (let i = 2; i < n; i += 1) {
        const t = dp[0];
        dp[0] = dp[1];
        dp[1] = (dp[0] + t) % 1000000007;
    }
    return dp[1];
}